pkg/controllers/management/node/controller.go

```diff
@@ -282,6 +282,20 @@ outer:
 }
 
 func (m *Lifecycle) Updated(obj *v3.Node) (*v3.Node, error) {
+	if isCustom(obj) {
+		m.setupCustom(obj)
+		newObj, err := v3.NodeConditionUpdated.Once(obj, func() (runtime.Object, error) {
```

Comment:
@aiwantaozi Can you check if Once would work for subsequent updates? From what I recall, Condition.Once means that the condition will be evaluated only once, and the callback will be called only once.

Reply:
Hi @alena1108 , if we return error in the once, just let the framework to retry, is no error return from the once, it will continue go to the subsequence to provision, is that ok?

Reply:
@aiwantaozi i think once means: once the condition is true, don't execute func(), which is not what we want in here?

Reply:
@StrongMonkey I thought so too. @aiwantaozi could you run a test when you update the node with roles 2 times? the second time after the first one is finished, and the condition is set to true.